plugins/pulp_ostree/plugins/db/model.py

```diff
@@ -0,0 +1,110 @@
+from datetime import datetime
+from hashlib import sha256
+
+from mongoengine import DateTimeField, StringField, DictField
+from pulp.server.db.model import SharedContentUnit
+
+from pulp_ostree.common import constants
+
+
+def generate_remote_id(url):
+    """
+    Generate a remote_id based on the url.
+
+    :param url: The remote URL.
+    :type url: basestring
+    :return: The generated ID.
+    :rtype:str
+    """
+    h = sha256()
+    h.update(url)
+    return h.hexdigest()
+
+
+class MetadataField(DictField):
+    """
+    Commit metadata.
+    """
+
+    def to_mongo(self, value):
```

Comment:
Interesting.
Depends on if we think we can substitute the dot (.)s back.  In order to do that reliably, we'd need to substitute the dot (.) with something that would never occur naturally in the original key.  I'm not sure it's possible or worth the effort.